### ISSUE TITLE
Skaffold 2.4.1, ko builder: Fails when docker isn't installed

### ISSUE THREAD

**Issue by Xjs** [GoogleContainerTools/skaffold#8786]
### Expected behavior

Skaffold build with the `ko` builder should work on a machine without Docker installed. This worked in 2.3.1.

### Actual behavior

Skaffold build with the `ko` builder fails saying: `daemon.Write response:  Build Failed. Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Check if docker is running.`

### Information

- Skaffold version: v2.4.1
- Operating system: macOS Ventura 13.3.1 (a)
- Installed via: Homebrew
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v2beta28 # doesn't work with v4beta5 either, I tried
kind: Config
build:
  artifacts:
    - image: eu.gcr.io/REDACTED/REDACTED
      ko:
        main: ./cmd/REDACTED
```

### Steps to reproduce the behavior

Try reproducing https://skaffold.dev/docs/builders/builder-types/ko/#configuring-the-ko-builder on a machine without Docker installed with v2.4.1. It used to work with v2.3.1.

### Logs from v2.3.1

```
Generating tags...
 - eu.gcr.io/REDACTED/REDACTED -> eu.gcr.io/REDACTED/REDACTED:v1.1.8-1-gREDACTED
Starting build...
Building [eu.gcr.io/REDACTED/REDACTED]...
WARN[0000] failed to list images: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?  subtask=eu.gcr.io/REDACTED/REDACTED task=Build
Using base distroless.dev/static:latest@sha256:8c5376baf48871a062e2d25e2107b783d518142d9d0c73bcd614d1b67d2d85bb for someredactedimportpath.com/redacted/cmd/redacted
Using build config eu.gcr.io/REDACTED/REDACTED for someredactedimportpath.com/redacted/cmd/redacted
Building someredactedimportpath.com/redacted/cmd/redacted for linux/amd64
Publishing eu.gcr.io/REDACTED/REDACTED:v1.1.8-1-g REDACTED
Published eu.gcr.io/REDACTED/REDACTED:v1.1.8-1-g REDACTED@sha256: REDACTED
Build [eu.gcr.io/REDACTED/REDACTED] succeeded
There is a new version (2.4.1) of Skaffold available. Download it from:
  https://github.com/GoogleContainerTools/skaffold/releases/tag/v2.4.1
```

### Logs from v2.4.1

```
Generating tags...
 - eu.gcr.io/REDACTED/REDACTED -> eu.gcr.io/REDACTED/REDACTED:v1.1.8-1-gREDACTED
Starting build...
Building [eu.gcr.io/REDACTED/REDACTED]...
WARN[0000] failed to list images: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?  subtask=eu.gcr.io/REDACTED/REDACTED task=Build
Using base cgr.dev/chainguard/static:latest@sha256:8c5376baf48871a062e2d25e2107b783d518142d9d0c73bcd614d1b67d2d85bb for someredactedimportpath.com/redacted/cmd/redacted
Using build config eu.gcr.io/REDACTED/REDACTED for someredactedimportpath.com/redacted/cmd/redacted
Building someredactedimportpath.com/redacted/cmd/redacted for linux/amd64
Loading eu.gcr.io/REDACTED/REDACTED:REDACTED
daemon.Write response:  
Build Failed. Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Check if docker is running.
Exit code: 1
```

**Comment by Xjs** [comment]
Any updates on this?

**Comment by andmn** [comment]
Just a hint, for Fedora 38 and skaffold ko builder you can use podman as non root:
1. `systemctl --user enable podman.socket`  => the podman service will start if required
2. set `export DOCKER_HOST=unix:///run/user/$UID/podman/podman.sock` and overwrite the default docker.sock

Works with skaffold 2.4.1 and 2.5.0.

**Comment by Xjs** [comment]
Thanks for the hint @andmn, however I don't want to use Podman instead of Docker, I don't see why any of these services is at all necessary to produce a .tar file containing a Go binary. That's exactly why I'm using `ko`, to simply build and push an image file without any shenanigans.

**Comment by Xjs** [comment]
Fixed in #8845, thanks @halvards!
